storage/innobase/handler/ha_innodb.cc

```diff
@@ -5633,6 +5633,20 @@ check_index_consistency(const TABLE* table, const dict_table_t* ib_table)
 			ret = false;
 			goto func_exit;
 		}
+
+		if (index->is_unique()) {
+			ulint n_index= 0;
+			for (dict_index_t *idx =
+				UT_LIST_GET_FIRST(ib_table->indexes);
+			     idx; idx= UT_LIST_GET_NEXT(indexes, idx)) {
+
+				if (idx == index && count != n_index) {
+					m_int_table_flags|=
+						HA_DUPLICATE_KEY_NOT_IN_ORDER;
+				}
```

Comment:
This is missing a `break;` or `return;`. The check is too strict as well. Any mismatch regarding the positions of non-unique indexes should not matter.